src/pybind/mgr/dashboard/frontend/cypress/integration/orchestrator/01-hosts-force-maintenance.e2e-spec.ts

```diff
@@ -25,8 +29,14 @@ describe('Hosts page', () => {
       });
 
       let enterMaintenance = true;
+
+      // Force maintenance might throw out error if host are less than 3.
+      if (hostnameList.length < 3) {
+        enterMaintenance = false
+      }
```

Comment:
This actually means that we need more than 3 hosts to test this part of the code?

Shouldn't this be handled by the host maintenance feature itself, rather than explicitly refraining ourselves from testing this?

If not, I'd prefer to make the test pre-conditions explicit upfront. Usually test frameworks have helpers for this: [`skipOn` or `onlyOn`](https://www.npmjs.com/package/@cypress/skip-test)

Reply:
I put this just for the testing purpose and ensure that force maintenance happens successfully. For host maintenance this is not a requirement, but for testing this I need to ensure this because I am checking if force maintenance modal is showing in the following steps and if there is an error from orch that won't be showing. So I am just ensuring that this host can be put into force maintenance.
I'll use these test frameworks.

Reply:
Its a little trickier to test the force maintenance. Because in some cases (for eg: if there is only 1 host) the force maintenance can't even be doable. So instead of showing the modal for forcing it, it just shows the error. And the test expects for a modal. that's why this error happens. So only thing to do is to make sure the environment has more than 2 hosts and it only has the services which allows hosts to go in maintenance. @epuertat